wpilibcIntegrationTests/src/FRCUserProgram/cpp/RobotDriveTest.cpp

```diff
@@ -0,0 +1,135 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2014-2017 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "Drive/DifferentialDrive.h"
+#include "MockSpeedController.h"
+#include "PIDController.h"
+#include "PIDOutput.h"
+#include "PIDSource.h"
+#include "RobotDrive.h"
+#include "TestBench.h"
+#include "Timer.h"
+#include "gtest/gtest.h"
+
+using namespace frc;
+
+class RobotDriveTest : public testing::Test {
+ protected:
+  MockSpeedController* m_rdFrontLeft;
+  MockSpeedController* m_rdRearLeft;
+  MockSpeedController* m_rdFrontRight;
+  MockSpeedController* m_rdRearRight;
+  MockSpeedController* m_frontLeft;
+  MockSpeedController* m_rearLeft;
+  MockSpeedController* m_frontRight;
+  MockSpeedController* m_rearRight;
+  RobotDrive* m_robotDrive;
+  DifferentialDrive* m_differentialDrive;
+
+  double m_testJoystickValues[7] = {-1.0, -0.5, -0.01, 0.0, 0.01, 0.5, 1.0};
+  double m_testGyroValues[19] = {0,    45,   90,   135,  180, 225,  270,
+                                 305,  360,  540,  -45,  -90, -135, -180,
+                                 -225, -270, -305, -360, -540};
+
+  void SetUp() override {
+    m_rdFrontLeft = new MockSpeedController();
```

Comment:
Java reuses these between tests. To have the same behavior, you could allocate the C++ objects inline with the declarations on the stack. For example:
```
MockSpeedController m_rdFrontLeft;
MockSpeedController m_rdRearLeft;
MockSpeedController m_rdFrontRight;
MockSpeedController m_rdRearRight;
frc::RobotDrive m_robotDrive{m_rdFrontLeft, m_rdRearLeft, m_rdFrontRight, m_rdRearRight};

frc::DifferentialDrive m_differentialDrive{m_frontLeft, m_frontRight};
```